File: Exam3/Exam3/FizzBuzzConverter.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Exam3
{
	class FizzBuzzConverter
	{
		public List<string> ConvertNumber(List<int> numbers)
		{
			var result = new List<string>();

			foreach (var number in numbers)
			{
				if (isFizz(number) && isBuzz(number))
				{
					result.Add("FizzBuzz");
				}
				else if (isFizz(number))
				{
					result.Add("Fizz");
				}
				else if (isBuzz(number))
				{
					result.Add("Buzz");
				}
				else
				{
					result.Add(number.ToString());
				}
			}

			return result;

		}

		private static bool isFizz(int currentCount)
		{
			return currentCount % 3 == 0;
		}


		private static bool isBuzz(int currentCount)
		{
			return currentCount % 5 == 0;
		}
	}
}

File: src/ch/adesso/exam3/FizzBuzzConverter.java
package ch.adesso.exam3;

import java.util.ArrayList;
import java.util.List;

public final class FizzBuzzConverter {
    public static List<String> convert(List<Integer> number){
        List<String> result = new ArrayList<>();
        for (Integer integer: number){
            if(integer % 5 == 0 && integer % 3 == 0)
                result.add("FizzBuzz");
            else if(integer % 5 == 0)
                result.add("Fizz");
            else if (integer % 3 == 0)
                result.add("Buzz");
            else
                result.add(integer.toString());
        }
        return result;
    }
}
